src/buildApplication/processVercelOutput.ts

```diff
@@ -27,6 +29,67 @@ export async function getVercelStaticAssets(): Promise<string[]> {
 	);
 }
 
+/**
+ * Copies the static assets from the default Vercel output directory to the custom output directory.
+ *
+ * @param outputDir Output directory to copy static assets to.
+ * @param vercelDir Default Vercel output directory.
+ * @param staticAssets List of static asset paths.
+ */
+export async function copyVercelStaticAssets(
+	outputDir: string,
+	vercelDir: string,
+	staticAssets: string[]
+): Promise<void> {
+	if (staticAssets.length === 0) return;
+	const plural = staticAssets.length > 1 ? 's' : '';
+	cliLog(
+		`Copying ${staticAssets.length} static asset${plural} to output directory...`
+	);
+
+	await Promise.all(
+		staticAssets.map(async file => {
+			const src = join(vercelDir, file);
+			const dest = join(outputDir, file);
+			try {
+				await mkdir(dirname(dest), { recursive: true });
+				await copyFile(src, dest);
+			} catch (e) {
+				cliError(`Failed to copy static asset '${file}' to output directory.`);
+			}
+		})
+	);
+}
+
+/**
+ * Prepares the custom output directory for the worker and static assets.
+ *
+ * Deletes the custom directory and its contents if it already exists.
+ *
+ * @param outputDir Custom output directory.
+ * @param staticAssets List of static asset paths.
+ */
+export async function setupOutputDir(
+	outputDir: string,
+	staticAssets: string[]
+) {
+	const vercelDir = normalizePath(resolve('.vercel', 'output', 'static'));
+
+	// If the output directory is not the default Vercel one, delete it if exists and create a new one.
+	// Then, copy the static assets from the default Vercel output directory to the new one.
+	if (!outputDir.startsWith(vercelDir)) {
+		cliLog(`Using custom output directory: ${relative('.', outputDir)}`);
+
+		await rm(outputDir, { recursive: true, force: true });
+		await mkdir(outputDir, { recursive: true });
+		await copyVercelStaticAssets(outputDir, vercelDir, staticAssets);
+
+		await deleteNextTelemetryFiles(outputDir);
```

Comment:
The approach I'm taking is to leave Vercel's output completely unaltered when you use a custom output directory, hence why I changed the deletion to be after all static assets are copied instead of before.

Reply:
That's a fair point, sure